core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java

```diff
@@ -262,4 +294,20 @@ private ChildProfileCredentialsProviderFactory stsCredentialsProviderFactory() {
             throw new IllegalStateException("Failed to create the '" + name + "' profile credentials provider.", e);
         }
     }
+
+    /**
+     * Load the factory that can be used to create the SSO credentials provider, assuming it is on the classpath.
+     */
+    private SsoCredentialsProviderFactory ssoCredentialsProviderFactory() {
+        try {
+            Class<?> ssoCredentialsProviderFactory = Class.forName(SSO_PROFILE_CREDENTIALS_PROVIDER_FACTORY, true,
+                                                                   Thread.currentThread().getContextClassLoader());
```

Comment:
You should check with John to see if this line is correct, since he's changing this line for STS.

Reply:
out of interest why are you doing custom class loading?

Reply:
What's the alternative? SSO isn't necessarily on the classpath.